Makefile

```diff
@@ -197,41 +225,68 @@ check_cert_manager:
                fi;\
        fi
 
+# Location to install dependencies
+LOCALBIN ?= $(shell pwd)/bin
+.PHONY: local-bin
+local-bin:
+	mkdir -p $(LOCALBIN)
+
+# Location to install test dependencies
+TESTBIN ?= $(shell pwd)/testbin
+.PHONY: test-bin
+test-bin:
+	mkdir -p $(TESTBIN)
+
 # Download controller-gen locally if necessary
-CONTROLLER_GEN = $(shell pwd)/bin/controller-gen
-controller-gen:
-	$(call go-get-tool,$(CONTROLLER_GEN),sigs.k8s.io/controller-tools/cmd/controller-gen@v0.4.1)
+CONTROLLER_GEN = $(LOCALBIN)/controller-gen
+.PHONY: controller-gen
+controller-gen: $(CONTROLLER_GEN)
+$(CONTROLLER_GEN): local-bin
+	[ -f $(CONTROLLER_GEN) ] || GOBIN=$(LOCALBIN) go install sigs.k8s.io/controller-tools/cmd/controller-gen@v$(CONTROLLER_GEN_VERSION)
 
 # Download kustomize locally if necessary
-KUSTOMIZE = $(shell pwd)/bin/kustomize
-kustomize:
-	$(call go-get-tool,$(KUSTOMIZE),sigs.k8s.io/kustomize/kustomize/v3@v$(KUSTOMIZE_VERSION))
+KUSTOMIZE_INSTALL_SCRIPT ?= "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh"
+KUSTOMIZE = $(LOCALBIN)/kustomize
+.PHONY: kustomize
+kustomize: $(KUSTOMIZE)
+$(KUSTOMIZE): local-bin
+	[ -f $(KUSTOMIZE) ] || { curl -s $(KUSTOMIZE_INSTALL_SCRIPT) | bash -s -- $(subst v,,$(KUSTOMIZE_VERSION)) $(LOCALBIN); }
 
 # Download addlicense locally if necessary
-ADDLICENSE = $(shell pwd)/bin/addlicense
-addlicense:
-	$(call go-get-tool,$(ADDLICENSE),github.com/google/addlicense@v$(ADDLICENSE_VERSION))
-
-# go-get-tool will 'go get' any package $2 and install it to $1.
-PROJECT_DIR := $(shell dirname $(abspath $(lastword $(MAKEFILE_LIST))))
-define go-get-tool
-@[ -f $(1) ] || { \
-set -e ;\
-TMP_DIR=$$(mktemp -d) ;\
-cd $$TMP_DIR ;\
-go mod init tmp ;\
-echo "Downloading $(2)" ;\
-GOBIN=$(PROJECT_DIR)/bin go get $(2) ;\
-rm -rf $$TMP_DIR ;\
-}
-endef
+ADDLICENSE = $(LOCALBIN)/addlicense
+.PHONY: addlicense
+addlicense: $(ADDLICENSE)
+$(ADDLICENSE): local-bin
+	[ -f $(ADDLICENSE) ] || GOBIN=$(LOCALBIN) go install github.com/google/addlicense@v$(ADDLICENSE_VERSION)
+
+# Download setup-envtest locally if necessary
+ENVTEST = $(TESTBIN)/setup-envtest
+.PHONY: setup-envtest
+setup-envtest: $(ENVTEST)
+$(ENVTEST): test-bin
+	[ -f $(ENVTEST) ] || GOBIN=$(TESTBIN) go install sigs.k8s.io/controller-runtime/tools/setup-envtest@latest
```

Comment:
With the target changed to e.g. `$(KUSTOMIZE)` the `-f` check should no longer be necessary. This target will only run if the file does not exist. We should also take the same approach for the `local-bin` target. Replace the `local-bin` target with a `$(LOCALBIN)` target. We could get rid of `test-bin` as well and just use `$(LOCALBIN)` for everything.

This section should be very similar to the boilerplate here (with the addition of the `addlicense` tool):
https://github.com/operator-framework/operator-sdk/blob/da3346113a8a75e11225f586482934000504a60f/testdata/go/v3/memcached-operator/Makefile#L152-L180

Sorry for fussing over the Makefile syntax, but the closer we stay to the boilerplate in the Makefile, the easier future Operator SDK migrations will be.

Reply:
Oh yess suree! Let me change that! I didn't know using `$(KUSTOMIZE)` will also do the check! Good to learn that!